Loop/Managers/Alerts/AlertStore.swift

```diff
@@ -65,20 +69,46 @@ extension AlertStore {
     }
     
     public func recordAcknowledgement(of identifier: Alert.Identifier, at date: Date = Date(),
-                                      completion: ((Result<Void, Error>) -> Void)? = nil) {
-        recordUpdateOfLatest(of: identifier, with: { $0.acknowledgedDate = date }, completion: completion)
+                                      _ completion: ((Result<Void, Error>) -> Void)? = nil) {
+        recordUpdateOfLatest(of: identifier, field: "acknowledgedDate", with: { $0.acknowledgedDate = date }, completion)
     }
     
     public func recordRetraction(of identifier: Alert.Identifier, at date: Date = Date(),
-                                 completion: ((Result<Void, Error>) -> Void)? = nil) {
-        recordUpdateOfLatest(of: identifier, with: { $0.retractedDate = date }, completion: completion)
+                                 _ completion: ((Result<Void, Error>) -> Void)? = nil) {
+        recordUpdateOfLatest(of: identifier, field: "retractedDate", with: { $0.retractedDate = date }, completion)
     }
     
+    public func lookupAllUnacknowledged(completion: @escaping (Result<[StoredAlert], Error>) -> Void) {
+        managedObjectContext.perform {
+            do {
+                let fetchRequest: NSFetchRequest<StoredAlert> = StoredAlert.fetchRequest()
+                fetchRequest.predicate =  NSCompoundPredicate(andPredicateWithSubpredicates: [
+                    NSPredicate(format: "acknowledgedDate == nil"),
+                    NSPredicate(format: "retractedDate == nil")
+                ])
+                fetchRequest.sortDescriptors = [ NSSortDescriptor(key: "modificationCounter", ascending: true) ]
```

Comment:
It seems to me there should be a priority to these alerts and that should be the primary sort field. For example, I'd like it would be more important to know about an urgent low blood glucose than then fact that my pump is going to expire in 8 hours.

Reply:
@darinkrauss we don't have a prioritization of alerts defined yet.  Until then, I can't make assumptions about them.